modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -326,299 +133,23 @@ function AddFDSFilterModalContent({
-									selectedClientExtension
-								}
-							/>
-						)}
-
-						{filterType === EFilterType.DATE_RANGE && (
-							<DateRangeFilterModalContent.Body
-								from={from}
-								isValidDateRange={isValidDateRange}
-								namespace={namespace}
-								onFromChange={setFrom}
-								onToChange={setTo}
-								onValidDateChange={(values) => {
-									setIsValidDateRange(values);
-									setSaveButtonDisabled(
-										isFormInvalid({
-											i18nFilterLabels,
-											isValidDateRange: values,
-											selectedClientExtension,
-											selectedField,
-											selectedPicklist,
-											sourceType,
-										})
-									);
-								}}
-								to={to}
-							/>
-						)}
-
-						{filterType === EFilterType.SELECTION && (
-							<SelectionFilterModalContent.Body
-								includeMode={includeMode}
-								multiple={multiple}
-								namespace={namespace}
-								onIncludeModeChange={setIncludeMode}
-								onMultipleChange={setMultiple}
-								onPreselectedValuesChange={(values) => {
-									setPreselectedValues(values);
-
-									setIncludeMode(
-										values.length
-											? filter &&
-											  (filter as ISelectionFilter)
-													.include
-												? 'include'
-												: 'exclude'
-											: 'include'
-									);
-								}}
-								onSelectedPicklistChange={(
-									values: IPickList | undefined
-								) => {
-									setSelectedPicklist(values);
-									setSaveButtonDisabled(
-										isFormInvalid({
-											i18nFilterLabels,
-											isValidDateRange,
-											selectedClientExtension,
-											selectedField,
-											selectedPicklist: values,
-											sourceType,
-										})
-									);
-								}}
-								onSourceChange={(values) => {
-									setSourceType(values);
-									setSaveButtonDisabled(
-										isFormInvalid({
-											i18nFilterLabels,
-											isValidDateRange,
-											selectedClientExtension,
-											selectedField,
-											selectedPicklist,
-											sourceType: values,
-										})
-									);
-								}}
-								picklists={picklists}
-								preselectedValues={preselectedValues}
-								selectedPicklist={selectedPicklist}
-								sourceType={sourceType}
-							/>
-						)}
-					</>
-				)}
-			</ClayModal.Body>
-
-			<ClayModal.Footer
-				last={
-					<ClayButton.Group spaced>
-						<ClayButton
-							disabled={saveButtonDisabled}
-							onClick={handleFilterSave}
-							type="submit"
-						>
-							{Liferay.Language.get('save')}
-						</ClayButton>
-
-						<ClayButton
-							displayType="secondary"
-							onClick={() => closeModal()}
-						>
-							{Liferay.Language.get('cancel')}
-						</ClayButton>
-					</ClayButton.Group>
-				}
+			<Component.Body
+				closeModal={closeModal}
+				fdsFilterClientExtensions={fdsFilterClientExtensions}
+				fieldNames={fieldNames}
+				fields={fields}
+				filter={filter}
+				handleSave={(body: any) => handleFilterSave(body)}
```

Comment:
The button in modal says `Save`, right? Then this prop should be `onSave`. As a general convention, we want to promote `on*` over `handle*`, for consistency. One is not better than the other, but `on*` is 10 times more common in DXP.

Reply:
Done